### PR TITLE
feat(protection): Separate interactions from ACE_MainActions and put them on corresponding members

### LINKED ISSUE #3: Add animation for equiping gear.

### LINKED ISSUE #7: Separate equiping actions from ACE_MainActions to ACE_Head and ACE_Torso

### LINKED ISSUE #9: Add an icon to distinguish the main interaction

### PR DESCRIPTION
**Pull request by PhileasLX**
#7

### CODE REVIEW

**Review by ArwynFr — COMMENTED** [review]
/delegate @ImplicitDeny 😄

File: addons/gsri_vip/functions/protection/fn_vipCheckForGear.sqf [hunk 1]
```diff
@@ -19,24 +26,30 @@ private _actions = [];
...
 			};
 		};
 
 		case "headgear": {
-			private _action = [format ["equip%1",_x], _displayName, "fr\gsri\vip\ui\icon_helmet.paa", GSRI_fnc_vipTryEquipGear, {true}, {}, _x] call ace_interact_menu_fnc_createAction;
-			_actions pushBack [_action, [], _target];
+			if(_bodyPart isEqualTo "head") then {
+				private _action = [format ["equip%1",_x], _displayName, "fr\gsri\vip\ui\icon_helmet.paa", GSRI_fnc_vipTryEquipGear, {true}, {}, _x] call ace_interact_menu_fnc_createAction;
+				_actions pushBack [_action, [], _target];
+			}
```
Comment (ImplicitDeny) [inline addons/gsri_vip/functions/protection/fn_vipCheckForGear.sqf]: Missing ;

File: addons/gsri_vip/functions/protection/fn_vipCheckForGear.sqf [hunk 2]
```diff
@@ -19,24 +26,30 @@ private _actions = [];
...
 		};
```
Comment (ImplicitDeny) [inline addons/gsri_vip/functions/protection/fn_vipCheckForGear.sqf]: Bad indent on case closing brackets

**Review by ImplicitDeny — CHANGES_REQUESTED** [review]
A few formatting mistakes to fix

File: addons/gsri_vip/functions/protection/fn_vipCheckForGear.sqf (hunk 1 above)
Comment (PhileasLX) [reply addons/gsri_vip/functions/protection/fn_vipCheckForGear.sqf]: It has been fixed.

File: addons/gsri_vip/functions/core/fn_initVip.sqf [hunk 3]
```diff
@@ -1,5 +1,9 @@
...
-	private _protectionMenu  = ["protectionMenu", localize "STR_VIP_protection", "", {}, {true}, GSRI_fnc_vipCheckForGear] call ace_interact_menu_fnc_createAction;
+    private _protectionMenu  = ["protectionMenu", localize "STR_VIP_protection", "\A3\ui_f\data\igui\cfg\simpleTasks\types\defend_ca.paa", {}, {true}, GSRI_fnc_vipCheckForGear, "head"] call ace_interact_menu_fnc_createAction;
+    ["CAManBase", 0, ["ACE_Head"], _protectionMenu, true] call ace_interact_menu_fnc_addActionToClass;
 
-	["CAManBase", 0, ["ACE_MainActions"], _protectionMenu, true] call ace_interact_menu_fnc_addActionToClass;
+	_protectionMenu  = ["protectionMenu", localize "STR_VIP_protection", "\A3\ui_f\data\igui\cfg\simpleTasks\types\defend_ca.paa", {}, {true}, GSRI_fnc_vipCheckForGear, "torso"] call ace_interact_menu_fnc_createAction;
+	["CAManBase", 0, ["ACE_Torso"], _protectionMenu, true] call ace_interact_menu_fnc_addActionToClass;
+
+	_changeSpeedCoeffEH = ["vipChangeSpeedCoeff", {params ["_target", "_coeff"]; _target setAnimSpeedCoef _coeff;}] call CBA_fnc_addEventHandler;
 };
```
Comment (ImplicitDeny) [inline addons/gsri_vip/functions/core/fn_initVip.sqf]: missing LF

File: addons/gsri_vip/functions/protection/fn_vipCheckForGear.sqf [hunk 4]
```diff
@@ -1,44 +1,59 @@
...
+        case "glasses": {
+            if (_bodyPart isEqualTo "head") then {
+                private _action = [format ["equip%1", _x], _displayGlassesName, "fr\gsri\vip\ui\icon_facewear.paa", GSRI_fnc_vipTryEquipGear, {true}, {}, _x] call ace_interact_menu_fnc_createAction;
+                _actions pushBack [_action, [], _target];
+            };
+        };
+    };
+} forEach (backpackItems _player);
 
 _actions;
```
Comment (ImplicitDeny) [inline addons/gsri_vip/functions/protection/fn_vipCheckForGear.sqf]: Missing LF

File: addons/gsri_vip/functions/protection/fn_vipEquipGear.sqf [hunk 5]
```diff
@@ -1,74 +1,74 @@
...
+    ],
+    [
+        "_gearToEquip", 
+        "", 
+        [""]
+    ]
 ];
 
-if (_gearToEquip isEqualTo "") exitWith { diag_log format["GSRI VIP: vipEquipGear: no _gearToEquip."] };
+if (_gearToEquip isEqualTo "") exitWith { diag_log ["GSRI VIP: vipEquipGear: no _gearToEquip."] };
```
Comment (ImplicitDeny) [inline addons/gsri_vip/functions/protection/fn_vipEquipGear.sqf]: unnecessary [ ]

File: addons/gsri_vip/functions/protection/fn_vipEquipGearFailure.sqf [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	]
+];
+
+private _animationTransition = "amovp[pos]mstpsnonwnondnon";
+private _pos = ["erc", "knl", "pne"] param [["STAND", "CROUCH", "PRONE"] find stance _player];
+_animationTransition = [_animationTransition, "[pos]", _pos] call CBA_fnc_replace;
+[_player, _animationTransition, 2] call ace_common_fnc_doAnimation;
+[_player, _animName, 1] call ace_common_fnc_doAnimation;
+
+["vipChangeSpeedCoeff", [_player, 1]] call CBA_fnc_globalEvent;
```
Comment (ImplicitDeny) [inline addons/gsri_vip/functions/protection/fn_vipEquipGearFailure.sqf]: Missing LF

File: addons/gsri_vip/functions/protection/fn_vipEquipGearSuccess.sqf [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+];
+
+_args call GSRI_fnc_vipEquipGear;
+private _animationTransition = "amovp[pos]mstpsnonwnondnon";
+private _pos = ["erc", "knl", "pne"] param [["STAND", "CROUCH", "PRONE"] find stance _player];
+_animationTransition = [_animationTransition, "[pos]", _pos] call CBA_fnc_replace;
+[_player, _animationTransition, 2] call ace_common_fnc_doAnimation;
+[_player, _animName, 1] call ace_common_fnc_doAnimation;
+
+["vipChangeSpeedCoeff", [_player, 1]] call CBA_fnc_globalEvent;
```
Comment (ImplicitDeny) [inline addons/gsri_vip/functions/protection/fn_vipEquipGearSuccess.sqf]: Missing LF

File: addons/gsri_vip/functions/protection/fn_vipTryEquipGear.sqf [hunk 8]
```diff
@@ -1,25 +1,56 @@
...
+	_animCfgSpeed = getNumber (configfile >> "CfgMovesMaleSdr" >> "States" >> _x >> "speed");
+	if(_animCfgSpeed != 0) then {_animDuration = _animDuration + abs (1 / _animCfgSpeed);};	
+} forEach _animArray;
+
+private _animRatio = _animDuration / _duration;
+
+["vipChangeSpeedCoeff", [_player, _animRatio]] call CBA_fnc_globalEvent;
+[_player, "Act_Alien_Gesture"] call ace_common_fnc_doAnimation;
+
+[_duration, [_target, _player, _gear, _currentAnimName] , GSRI_fnc_vipEquipGearSuccess, GSRI_fnc_vipEquipGearFailure, localize "STR_VIP_settingUpProtection"] call ace_common_fnc_progressBar;
```
Comment (ImplicitDeny) [inline addons/gsri_vip/functions/protection/fn_vipTryEquipGear.sqf]: Missing LF

**Review by ImplicitDeny — CHANGES_REQUESTED** [review]
A few fixes

**Review by ImplicitDeny — APPROVED** [review]

**Pull request merged**